docs/docs/reference/contextual/derivation.md

```diff
@@ -284,16 +285,30 @@ object Eq {
         }
     }
 
-  inline given derived[T](given m: Mirror.Of[T]): Eq[T] = {
-    val elemInstances = summonAll[m.MirroredElemTypes]
-    inline m match {
-      case s: Mirror.SumOf[T]     => eqSum(s, elemInstances)
-      case p: Mirror.ProductOf[T] => eqProduct(p, elemInstances)
+  inline def derived[T]: Eq[T] =
+    summonFrom {
+      case m: Mirror.Of[T] =>
+        val elemInstances = summonAll[m.MirroredElemTypes]
+        inline m match {
+          case s: Mirror.SumOf[T]     => eqSum(s, elemInstances)
+          case p: Mirror.ProductOf[T] => eqProduct(p, elemInstances)
+        }
     }
-  }
 }
 ```
 
+> Note that we can explicitly state the dependence on `Mirror[T]` in `derived` by using the
+> following encoding:
+> ```scala
+> inline def derived[T](given m: Mirror.Of[T]): Eq[T]
+> ```
+> This is also equivalent to using the Implicit Function Type:
```

Comment:
I have no real objection to this, but it would be nice if you could elaborate on why you think the variant which removes the given from the signature of `derived` is preferred ... on the face of it both less declarative and more verbose, so why switch?

Reply:
I don't know what the better way is. The whole reason I went with this is so that I could show how it's possible to do the `inline implicit def autoEq[T]: Eq[T] = Eq.derived` approach but I know that's not ideal. I have to go in this direction for Quill's use-case because it's not reasonable to suddenly expect all my users to add `derives GenericDecoder` to all of their entity case-classes since they don't do that now. I didn't know that you could do inline `given derived[T]: Mirror.Of[T] ?=> Eq[T]` like Aggelos suggested and solve this. I'll have to try that out.

The issue is that there are 3 different ways to do this stuff, I think that all 3 should at least be mentioned since different people need different things.